### ISSUE TITLE
TypeError: expected bytes, str found

### ISSUE THREAD

**Issue by biaken** [grpc/grpc#16718]
### What version of gRPC and what language are you using?
 grpcio, and grcpio-tools version: 1.15.0
python

 
### What operating system (Linux, Windows, …) and version?
Problem encountered both in Centos 7.5 & Python 3.6.5 (Docker setup) as well as; MacOS High Sierra 10.13.6 & python: 3.6.2
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 python 3.6
 
### What did you do?
Program: A grpc ssl helloworld greeter; server and client python scripts

How i generate my proto:
```python -m grpc_tools.protoc --python_out=. --grpc_python_out=. --proto_path=. helloworld.proto```

How i generate self signed certs:
```openssl req -x509 -newkey rsa:4096 -keyout self.key -out self.crt -days 365  -nodes -subj '/CN=localhost'```

Client Code:
```
def run_secure_crt():
    print('Secure Client call (crt)')
    private_crt_file = 'self.crt'
    trusted_certs = open(private_crt_file).read()
    credentials = grpc.ssl_channel_credentials(root_certificates=trusted_certs)
    channel = grpc.secure_channel('localhost:16506', credentials)

    stub = helloworld_pb2_grpc.GreeterStub(channel)

    response = stub.SayHello(helloworld_pb2.HelloRequest(name='you'))
    print("Greeter client received (crt): " + response.message)
```

Server Code:
```
def serve_ssl():
    print('Starting up grpc server (ssl)...', os.getpid())
    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
    helloworld_pb2_grpc.add_GreeterServicer_to_server(Greeter(), server)

    private_key_file = 'self.key'
    with open(private_key_file, 'rb') as f:
        private_key = f.read()

    cert_file = 'self.crt'
    with open(cert_file, 'rb') as f:
        cert_chain = f.read()

    server_cred = grpc.ssl_server_credentials(((private_key, cert_chain,),))
    server.add_secure_port('[::]:16506', server_cred)

    server.start()
    print('Started at', 'localhost:16506 (ssl)', os.getpid())
    try:
        while True:
            time.sleep(_ONE_DAY_IN_SECONDS)
    except KeyboardInterrupt:
        print('Stoping grpc server ', os.getpid(), 'localhost:16506')
        server.stop(0)
```

helloworld.proto
```
syntax = "proto3";

//option java_multiple_files = true;
//option java_package = "io.grpc.examples.helloworld";
//option java_outer_classname = "HelloWorldProto";
//option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
```

 
### What did you expect to see?
 A successful grpc call
 ```
$ python greeter_client.py 
Secure Client call (crt)
Greeter client received (crt): Hello, you!
```

### What did you see instead?
`TypeError: expected bytes, str found`
 ```
$ python greeter_client.py 
Secure Client call (crt)
TypeError: expected bytes, str found
Exception ignored in: 'grpc._cython.cygrpc.SSLChannelCredentials.c'
TypeError: expected bytes, str found
Traceback (most recent call last):
  File "greeter_client.py", line 61, in <module>
    run_secure_crt()
  File "greeter_client.py", line 41, in run_secure_crt
    response = stub.SayHello(helloworld_pb2.HelloRequest(name='you'))
  File "/Users/ginaulak/anaconda/envs/py35/lib/python3.6/site-packages/grpc/_channel.py", line 532, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/Users/ginaulak/anaconda/envs/py35/lib/python3.6/site-packages/grpc/_channel.py", line 466, in _end_unary_response_blocking
    raise _Rendezvous(state, None, None, deadline)
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "lame client channel"
	debug_error_string = "{"created":"@1538040595.935106000","description":"lame client channel","file":"src/core/lib/surface/lame_client.cc","file_line":90}"
>
```

### Anything else we should know about your project / environment?
I tried with the dev build mentioned in : https://github.com/grpc/grpc/issues/16623 Unfortunately, it does not work for me. I also tried with grpcio 1.8.6, did not work.

It works with grpcio 1.7.0 though. However, I also want to have Tensorflow 1.10 in my project which require grpcio>=1.8.6 
If i try with grpcio==1.7.0, it is working fine.
```
$ python greeter_client.py 
Secure Client call (crt)
Greeter client received (crt): Hello, you!
```

**Comment by mehrdada** [comment]
related to https://github.com/grpc/grpc/issues/16643

**Comment by lidizheng** [comment]
@biaken Hi,

The quick fix for this is read the client-side credential with `rb` flag:
```
trusted_certs = open(private_crt_file, 'rb').read()
```
And it will work. As for exception surface for proper notification, as Mehrdad said, we are working on it at #16643.

**Comment by lidizheng** [comment]
In PR #16971, we believe there should be another fix separated from exception surfacing. An exception should be raised when the `grpc.ssl_channel_credentials` be called, instead of raising exception after passing credentials to `grpc.secure_channel`.

**Comment by lidizheng** [comment]
This issue should be fixed by PR #17281.
